osu.Game.Rulesets.Catch/Skinning/Legacy/CatchLegacySkinTransformer.cs

```diff
@@ -22,14 +24,17 @@ public CatchLegacySkinTransformer(ISkinSource source)
 
         public override Drawable GetDrawableComponent(ISkinComponent component)
         {
-            if (component is HUDSkinComponent hudComponent)
+            if (component is SkinnableTargetComponent targetComponent && targetComponent.Target == SkinnableTarget.MainHUDComponents)
             {
-                switch (hudComponent.Component)
-                {
-                    case HUDSkinComponents.ComboCounter:
-                        // catch may provide its own combo counter; hide the default.
-                        return providesComboCounter ? Drawable.Empty() : null;
-                }
+                if (!providesComboCounter || !(Source.GetDrawableComponent(component) is SkinnableTargetComponentsContainer components))
```

Comment:
This reads badly. Split it out or write it differently. As it stands the code following this line is very hard to make sense of.

Reply:
```csharp
if (!providesComboCounter)
    return null;
    
var components = Source.GetDrawableComponent(component) as SkinnableTargetComponentsContainer;
if (components == null)
    return null;
```
does this read better?

Reply:
The second line can probably still do the `is` assignment. It's just the fact you had it at the end of an `||` conditions makes it very hard to mentally understand.

Reply:
I see, noted.